lib/spack/spack/test/concretize.py

```diff
@@ -925,6 +925,26 @@ def test_compiler_constraint_with_external_package(
         assert s.external
         assert s.satisfies(expected)
 
+    @pytest.mark.regression('20976')
+    def test_compiler_in_nonbuildable_external_package(self):
+        packages_yaml = {
+            'external-common-openssl': {
+                'externals': [
+                    {'spec': 'external-common-openssl@1.1.1i%gcc', 'prefix': '/usr'}
+                ],
+                'buildable': False
+            }
+        }
+        spack.config.set('packages', packages_yaml)
+
+        s = Spec("external-common-python %clang").concretized()
+        for d in s.dependencies():
+            if d.name == 'external-common-openssl':
+                compiler = 'gcc'
+            else:
+                compiler = 'clang'
+            assert s.satisfies('^' + d.name + '%' + compiler)
```

Comment:
Can we pass spec strings of what we want to check for this external, and check it like:
```python
for constraint in expected_constraints:
    s.satisfies(constraint) 
```
 instead of constructing them on the fly?

Reply:
Done